openupgrade_scripts/scripts/mail/16.0.1.10/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,222 @@
+mail         / mail.template            / null_value (char)             : DEL
+mail         / mail.template            / sub_model_object_field (many2one): DEL relation: ir.model.fields
+mail         / mail.template            / sub_object (many2one)         : DEL relation: ir.model
+mail         / mail.template            / template_fs (char)            : NEW
+# NOTHING TO DO
+
+
+mail         / res.partner              / channel_ids (many2many)       : table is now 'mail_channel_member' ('mail_channel_partner')
+# DONE in pre-migration script when renaming the model
+
+---XML records in module 'mail'---
+NEW ir.actions.act_window: mail.act_server_history [renamed from fetchmail module]
+NEW ir.actions.act_window: mail.action_email_server_tree [renamed from fetchmail module]
+NEW ir.actions.act_window: mail.mail_channel_member_action
+NEW ir.actions.act_window: mail.mail_gateway_allowed_action
+NEW ir.actions.act_window: mail.mail_link_preview_action
+NEW ir.actions.act_window: mail.mail_message_schedule_action
+NEW ir.actions.act_window: mail.mail_template_reset_action
+DEL ir.actions.act_window: mail.mail_channel_partner_action
+DEL ir.actions.act_window: mail.mail_resend_cancel_action
+# NOTHING TO DO: noupdate="0" records
+
+NEW ir.cron: mail.ir_cron_mail_gateway_action [renamed from fetchmail module] (noupdate)
+# NOTHING TO DO: modules merged in apriori.py
+
+NEW ir.cron: mail.ir_cron_send_scheduled_message (noupdate)
+# NOTHING TO DO: New noupdate="1" records
+
+NEW ir.model.access: mail.access_fetchmail_server [renamed from fetchmail module]
+NEW ir.model.access: mail.access_mail_channel_member_portal
+NEW ir.model.access: mail.access_mail_channel_member_public
+NEW ir.model.access: mail.access_mail_channel_member_user
+NEW ir.model.access: mail.access_mail_gateway_allowed_system
+NEW ir.model.access: mail.access_mail_link_preview_admin
+NEW ir.model.access: mail.access_mail_message_scheduled_all
+NEW ir.model.access: mail.access_mail_message_scheduled_system
+NEW ir.model.access: mail.access_mail_template_reset
+DEL ir.model.access: mail.access_mail_channel_partner_portal
+DEL ir.model.access: mail.access_mail_channel_partner_public
+DEL ir.model.access: mail.access_mail_channel_partner_user
+DEL ir.model.access: mail.access_mail_resend_cancel
+NEW ir.model.constraint: mail.constraint_mail_activity_check_res_id_is_set
+NEW ir.model.constraint: mail.constraint_mail_channel_channel_type_not_null
+NEW ir.model.constraint: mail.constraint_mail_channel_group_public_id_check
+NEW ir.model.constraint: mail.constraint_mail_channel_member_partner_or_guest_exists
+NEW ir.model.constraint: mail.constraint_mail_channel_rtc_session_channel_member_unique
+NEW ir.model.constraint: mail.constraint_res_users_notification_type
+DEL ir.model.constraint: mail.constraint_mail_channel_partner_partner_or_guest_exists
+DEL ir.model.constraint: mail.constraint_mail_channel_rtc_session_channel_partner_unique
```

Comment:
use delete_sql_constraint_safely method in pre-migration

Reply:
Done